### PR TITLE
[REVIEW] Modify reset_index to handle a new MC column

### LINKED ISSUE #3846: [BUG] Cudf looses column names after reset_index on  multi-column grouped frames
**Describe the bug**
Cudf looses column names after reset_index on  multi-column grouped frames.  This works for single column frames. 

**Steps/Code to reproduce bug**
```python
import cudf

df = cudf.DataFrame({'a':[1,2,3,4],'b':[5,1,2,5]})
df_groups = df.groupby(['a']).agg({'b': ['count', 'mean']})
df_groups = df_groups.reset_index()
print(df_groups.columns)
```

```python
Index(['a', 0, 1], dtype='object')
```
**Expected behavior**
```python
Index(['a', ('b', 'count'), ('b', 'mean')], dtype='object')
```
**Environment overview (please complete the following information)**
 - Method of cuDF install: [conda]

```
# packages in environment at /raid/vjawa/conda_install/conda_env/envs/cudf_12_20_jan_even:
cudf                      0.12.0b200120         py37_1712    rapidsai-nightly
dask-cudf                 0.12.0b200120         py37_1712    rapidsai-nightly
libcudf                   0.12.0b200120     cuda10.1_1712    rapidsai-nightly
```

**Additional context**

It works well with single columns and also causes issues in downstream dask-cudf  group-bys. 
 
```python
import cudf

df = cudf.DataFrame({'a':[1,2,3,4],'b':[5,1,2,5]})
df_groups = df.groupby(['a']).agg({'b': ['count']})
df_groups = df_groups.reset_index()
print(df_groups.columns)
```
```
Index(['a', 'b'], dtype='object')
```

### PR DESCRIPTION
**Pull request by thomcom**
Fixes #3846 by creating a new MultiIndex column when `reset_index` is used with a MultiColumn dataframe.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -1488,6 +1488,16 @@ def reset_index(self, drop=False, inplace=False):
                 out[name] = self.index._values
             for col_name, col_value in self._data.items():
                 out[col_name] = col_value
+            if isinstance(self.columns, cudf.core.multiindex.MultiIndex):
```
Comment (kkraus14) [inline python/cudf/cudf/core/dataframe.py]: We really need to make this a Pandas MultiIndex in the future... but in a future PR

File: python/cudf/cudf/tests/test_multiindex.py [hunk 2]
```diff
@@ -709,6 +709,14 @@ def test_multiindex_groupby_reset_index():
     assert_eq(pdg.reset_index(), gdg.reset_index())
 
 
+def test_multicolumn_reset_index():
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_multiindex.py]: Let's add a situation where we have a multi column grouping key so that we get a MultiIndex as well.

Comment (thomcom) [reply python/cudf/cudf/tests/test_multiindex.py]: done below

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (thomcom) [reply python/cudf/cudf/core/dataframe.py]: Are you wanting to drop `cudf.MultiIndex` entirely, or just when used as `columns`?

Comment (kkraus14) [reply python/cudf/cudf/core/dataframe.py]: Just when used as columns. We need to access these on the host as opposed to on the device so it doesn't make sense to use an on device MultiIndex.

Future PR though as this handling of multiindex for columns is kinda fragile as it is.

**Pull request merged**